internal/printer/printer.go

```diff
@@ -272,18 +277,38 @@ func (p *printer) printFuncSuffix(opts transform.TransformOptions, n *astro.Node
 	if n.Transition {
 		propagationArg = "'self'"
 	}
+	fmt.Printf("PROPAGATION ARG %v\n", propagationArg)
 	p.println(fmt.Sprintf("}, %s, %s);", filenameArg, propagationArg))
 	p.println(fmt.Sprintf("export default %s;", componentName))
 }
 
+var skippedAttributes = map[string]bool{
+	"define:vars":        true,
```

Comment:
These are exactly the same except that the object form doesn't skip `define:vars`.